centreon/www/class/config-generate/AgentConfiguration.php

```diff
@@ -104,12 +108,33 @@ private function formatOtelConfiguration(array $data, ConnectionModeEnum $connec
      */
     private function formatCmaConfiguration(array $data, ConnectionModeEnum $connectionMode): array
     {
+        $tokens = $this->readTokenRepository->findByNames(
+            array_map(
+                static fn(array $token): string => $token['name'],
+                $data['tokens']
+            )
+        );
+
+        $tokens = array_filter(
+            $tokens,
+            static fn(Token $token): bool =>  !(
+                $token->isRevoked()
+                || ($token->getExpirationDate() !== null && $token->getExpirationdate() < new \DateTimeImmutable())
+            )
+        );
         $configuration = [
             'otel_server' => $this->formatOtelConfiguration($data, $connectionMode),
             'centreon_agent' => [
                 'check_interval' => CmaConfigurationParameters::DEFAULT_CHECK_INTERVAL,
                 'export_period' => CmaConfigurationParameters::DEFAULT_EXPORT_PERIOD,
-            ]
+            ],
+            'tokens' => array_map(
+                static fn(JwtToken $token): array => [
+                    'token' => $token->getToken(),
+                    'encoding_key' => $token->getEncodingKey(),
+                ],
+                $tokens
```

Comment:
```suggestion
    'tokens' => array_map(
        static fn(JwtToken $token): array => [
            'token' => $token->getToken(),
            'encoding_key' => $token->getEncodingKey(),
        ],
        array_filter(
            $tokens,
            static fn(Token $token): bool =>  !(
                $token->isRevoked()
                || ($token->getExpirationDate() !== null && $token->getExpirationdate() < new \DateTimeImmutable())
            )
        )
    ),
```

Just a suggestion, you can resolve if you don't want, move the array filter directly hereso we don't assign variable that are just used once